minifi/minifi-toolkit/minifi-toolkit-configuration/src/main/java/org/apache/nifi/minifi/toolkit/configuration/json/TransformCommandFactory.java

```diff
@@ -0,0 +1,117 @@
+
+    private final PathInputStreamFactory pathInputStreamFactory;
+    private final PathOutputStreamFactory pathOutputStreamFactory;
+
+    public TransformCommandFactory(PathInputStreamFactory pathInputStreamFactory,
+            PathOutputStreamFactory pathOutputStreamFactory) {
+        this.pathInputStreamFactory = pathInputStreamFactory;
+        this.pathOutputStreamFactory = pathOutputStreamFactory;
+    }
+
+    public ConfigMain.Command create() {
+        return new ConfigMain.Command(this::transformToJson, COMMAND_DESCRIPTION);
+    }
+
+    private int transformToJson(String[] args) {
+        if (args.length != 3) {
+            printTransformUsage();
+            return ConfigMain.ERR_INVALID_ARGS;
+        }
+
+        String sourceNiFiJsonPath = args[1];
+        String targetMiNiFiJsonPath = args[2];
+
+        try {
+            RegisteredFlowSnapshot registeredFlowSnapshot = readNifiFlow(sourceNiFiJsonPath);
+            VersionedDataflow versionedDataflow = new VersionedDataflow();
+            versionedDataflow.setRootGroup(registeredFlowSnapshot.getFlowContents());
+            persistFlowJson(versionedDataflow, targetMiNiFiJsonPath);
+        } catch (ConfigTransformException e) {
+            System.out.println("Unable to convert NiFi JSON to MiNiFi flow JSON: " + e);
+            return e.getErrorCode();
+        }
+        return ConfigMain.SUCCESS;
+    }
+
+    private void printTransformUsage() {
+        System.out.println("Transform Usage:");
+        System.out.println();
+        System.out.println(" transform SOURCE_NIFI_JSON_FLOW_FILE TARGET_MINIFI_JSON_FLOW_FILE");
+        System.out.println();
+    }
+
+    private RegisteredFlowSnapshot readNifiFlow(String sourceNiFiJsonPath) throws ConfigTransformException {
+        try (InputStream inputStream = pathInputStreamFactory.create(sourceNiFiJsonPath)) {
+            ObjectMapper objectMapper = createObjectMapper();
+            final RegisteredFlowSnapshot registeredFlowSnapshot = objectMapper.readValue(inputStream,
+                    RegisteredFlowSnapshot.class);
+            return registeredFlowSnapshot;
+
+        } catch (IOException e) {
+            throw new ConfigTransformException("Error when reading NiFi flow json file",
+                    ConfigMain.ERR_UNABLE_TO_OPEN_INPUT, e);
+        }
+    }
+
+    private void persistFlowJson(VersionedDataflow flow, String flowJsonPath) throws ConfigTransformException {
+        try (OutputStream outputStream = pathOutputStreamFactory.create(flowJsonPath)) {
+            ObjectMapper objectMapper = createObjectMapper();
+            objectMapper.writeValue(outputStream, flow);
+        } catch (IOException e) {
+            throw new ConfigTransformException("Error when persisting flow JSON file: " + flowJsonPath,
+                    ConfigMain.ERR_UNABLE_TO_SAVE_CONFIG, e);
+        }
+    }
+
+    private ObjectMapper createObjectMapper() {
```

Comment:
This is a duplication of TransformCommandFactory.createObjectMapper method. We should create a utility class with moving the this method there and making it static. Then we could use it from TransformCommandFactory and TransformYamlCommandFactory, avoiding code duplication.